opencda/scenario_testing/rl_single_town06_carla_v2v.py

```diff
@@ -0,0 +1,314 @@
+from ding.utils import set_pkg_seed
+from ding.rl_utils import get_epsilon_greedy_fn
+
+import opencda.scenario_testing.utils.sim_api as sim_api
+from opencda.core.common.cav_world import CavWorld
+from opencda.scenario_testing.evaluations.evaluate_manager import EvaluationManager
+from opencda.scenario_testing.utils.yaml_utils import load_yaml, save_yaml
+from opencda.core.ml_libs.rl.envs.simple_carla_env_v2_scenario_manager import CarlaRLEnv
+from opencda.core.ml_libs.rl.utils.others.tcp_helper import parse_carla_tcp
+from opencda.core.ml_libs.rl.rl_models import DQNRLModel, TD3RLModel
+from opencda.core.ml_libs.rl.utils.others.ding_utils import compile_config
+from opencda.core.ml_libs.rl.utils.others.ding_utils import read_ding_config
+from opencda.core.ml_libs.rl.eval.single_carla_evaluator import SingleCarlaEvaluator
+from opencda.core.ml_libs.rl.eval.carla_benchmark_evaluator import CarlaBenchmarkEvaluator
+
+
+def init_carla_env(env_cfg, host, port, tm_port=None):
+    """
+    Initiate a Carla environment for rl agent.
+    Parameters
+    ----------
+    env_cfg:dict
+        The parsed configuration file.
+    host:str
+        The local host name.
+    port:int
+        The port for CARLA server.
+    tm_port:int (optional)
+        The tm port to run CARLA traffic manager. Default value is none as CARLA is able
+        to generate new TM port automatically.
+
+    Returns
+    -------
+
+    """
+    env = CarlaRLEnv(env_cfg, host, port, tm_port)
+    return env
+
+
+def get_rl_policy(spec):
+    """
+    Get the desired RL policy and model based on configuration file.
+    Parameters
+    ----------
+    spec:str
+        The specific name of the desired policy, should be parsed from config file.
+    Returns
+    -------
+    (policy_cls, model_cls): DI engine interfaces
+        The DI-engine RL policy and RL model interface.
+
+    """
+    policy_cls, model_cls = {
+        'dqn': (DQNPolicy, DQNRLModel),
+        'td': (TD3Policy, TD3RLModel)
+        # todo: add more models here
+    }[spec]
+    return policy_cls, model_cls
+
+
+def get_rl_cfg(opt, default_train_config):
+    """
+    Get the structured configurations for DI-engine's environment manager.
+    Parameters
+    ----------
+    opt:str
+        The terminal argument passed in from opencda.py.
+    default_train_config:dict
+        The Default configuration file.
+
+    Returns
+    -------
+    cfg:dict
+        The structured configuration file, matches the format requirement of environment manager.
+    """
+    # read config
+    default_train_config.exp_name = 'simple-rl'
+    opt_policy, _ = get_rl_policy(opt.policy)
+    # compile cfg
+    cfg = compile_config(
+        cfg=default_train_config,
+        env_manager=SyncSubprocessEnvManager,
+        policy=opt_policy,
+        learner=BaseLearner,
+        collector=SampleSerialCollector,
+        buffer=AdvancedReplayBuffer,
+    )
+    return cfg
+
+
+def rl_train(opt, config_yaml, seed=0):
```

Comment:
is this rl_train script specfic or more general? if it is general, consider put it in a seperate script.